### PR TITLE
Update patch versions and add new ones for all supported languages

### LINKED ISSUE #10466: Upgrade rust build environment to 1.70
## What's the problem this feature will solve?

The most recent currently available version of rust on RTD is 1.64, rust itself is at 1.70. In most cases 1.64 is fine, but there is a increasing amount of crates requiring newer features of the language, so repositories using those can not be build on RTD. 

The case I was running into this was when building https://github.com/nmandery/h3ronpy :

The build log output was 

```
Building wheels for collected packages: h3ronpy
  Building wheel for h3ronpy (pyproject.toml): started
  Building wheel for h3ronpy (pyproject.toml): finished with status 'error'
  error: subprocess-exited-with-error
  
  × Building wheel for h3ronpy (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [10 lines of output]
      Running `maturin pep517 build-wheel -i /home/docs/checkouts/readthedocs.org/user_builds/h3ronpy/envs/latest/bin/python --compatibility off`
      📦 Including license file "/home/docs/checkouts/readthedocs.org/user_builds/h3ronpy/checkouts/latest/LICENSE.txt"
      🍹 Building a mixed python/rust project
      🔗 Found pyo3 bindings with abi3 support for Python ≥ 3.8
      🐍 Not using a specific python interpreter
      error: package `konst v0.3.5` cannot be built because it requires rustc 1.65.0 or newer, while the currently active rustc version is 1.64.0
      💥 maturin failed
        Caused by: Failed to build a native library through cargo
        Caused by: Cargo build finished with "exit status: 101": `PYO3_ENVIRONMENT_SIGNATURE="cpython-3.9-64bit" PYO3_PYTHON="/home/docs/checkouts/readthedocs.org/user_builds/h3ronpy/envs/latest/bin/python" PYTHON_SYS_EXECUTABLE="/home/docs/checkouts/readthedocs.org/user_builds/h3ronpy/envs/latest/bin/python" "cargo" "rustc" "--message-format" "json-render-diagnostics" "--manifest-path" "/home/docs/checkouts/readthedocs.org/user_builds/h3ronpy/checkouts/latest/Cargo.toml" "--release" "--lib"`
      Error: command ['maturin', 'pep517', 'build-wheel', '-i', '/home/docs/checkouts/readthedocs.org/user_builds/h3ronpy/envs/latest/bin/python', '--compatibility', 'off'] returned non-zero exit status 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for h3ronpy
Failed to build h3ronpy
```

## Describe the solution you'd like

Having a more recent rust version available would be great. 1.65 would seemingly be sufficient for this issue, but upgrading directly to 1.70 would certainly be more sustainable.

I saw  #9692 and while I am not too familiar with how RTD works, I could try to submit a PR according to whats done there.

## Alternative solutions

I am not aware of any good workarounds.

## Additional context

-

### PR DESCRIPTION
**Pull request by EwoutH**
Update the Python patch versions of CPython 3.7 to 3.11 and PyPy 3.8 and 3.9 to their latest versions.

This PR is comparable to #9692 and needs https://github.com/readthedocs/readthedocs-docker-images/pull/192.

Closes #10466

### CODE REVIEW

**Comment by EwoutH** [comment]
Thanks for the review!

What's the current status?

**Comment by humitos** [comment]
No progress here 😞 . Unfortunately, I'm not able to prioritize this work currently due that I have a lot of other tasks that I need to finish.

**Comment by EwoutH** [comment]
No problem, thanks for the update!

File: readthedocs/settings/base.py [hunk 1]
```diff
@@ -622,14 +622,14 @@ def TEMPLATES(self):
...
-                '3.7': '3.7.15',
-                '3.8': '3.8.15',
-                '3.9': '3.9.15',
-                '3.10': '3.10.8',
-                '3.11': '3.11.0',
+                '3.7': '3.7.16',
+                '3.8': '3.8.16',
+                '3.9': '3.9.16',
+                '3.10': '3.10.10',
+                '3.11': '3.11.2',
```
Comment (humitos) [inline readthedocs/settings/base.py:629]:
```suggestion
                '3.7': '3.7.17',
                '3.8': '3.8.17',
                '3.9': '3.9.17',
                '3.10': '3.10.12',
                '3.11': '3.11.4',
```

File: readthedocs/settings/base.py [hunk 2]
```diff
@@ -622,14 +622,14 @@ def TEMPLATES(self):
...
                 'pypy3.7': 'pypy3.7-7.3.9',
-                'pypy3.8': 'pypy3.8-7.3.9',
-                'pypy3.9': 'pypy3.9-7.3.9',
+                'pypy3.8': 'pypy3.8-7.3.11',
+                'pypy3.9': 'pypy3.9-7.3.11',
```
Comment (humitos) [inline readthedocs/settings/base.py:632]:
```suggestion
                'pypy3.9': 'pypy3.9-7.3.12',
```

**Review by humitos — APPROVED** [review]
Upgrade the versions to match latest ones.

**Comment by humitos** [comment]
I'm trying to upload the `.tar.gz` to S3 via CircleCI in https://github.com/readthedocs/readthedocs-docker-images/issues/191. Once that's working, we can merge this PR.

**Comment by humitos** [comment]
All the new versions were upload successfully. I'm merging this PR. It will be deployed on Tuesday. Let us know if Rust 1.70 works for you after the deploy.

**Comment by nmandery** [comment]
Thank you for the very quick update - I will check Rust 1.70 after it has been deployed.

**Comment by stsewd** [comment]
We should update our docs https://github.com/readthedocs/readthedocs.org/blob/main/docs/user/config-file/v2.rst and spec https://github.com/readthedocs/readthedocs.org/blob/main/readthedocs/rtd_tests/fixtures/spec/v2/schema.json with the new versions.

**Pull request merged**